graphblas/io.py

```diff
@@ -220,6 +220,60 @@ def from_awkward(A, *, name=None):
     raise ValueError(f"Invalid format for Matrix: {format}")
 
 
+def from_pydata_sparse(s, *, dup_op=None, name=None):
+    """Create a Vector or a Matrix from a pydata.sparse array or matrix.
+
+    Input data in "gcxs" format will be efficient when importing with SuiteSparse:GraphBLAS.
+
+    Parameters
+    ----------
+    s : sparse
+        PyData sparse array or matrix (see https://sparse.pydata.org)
+    dup_op : BinaryOp, optional
+        Aggregation function for formats that allow duplicate entries (e.g. coo)
+    name : str, optional
+        Name of resulting Matrix
+
+    Returns
+    -------
+    :class:`~graphblas.Vector`
+    :class:`~graphblas.Matrix`
+    """
+    try:
+        import sparse
+    except ImportError:  # pragma: no cover (import)
+        raise ImportError("sparse is required to import from pydata sparse") from None
+    if not isinstance(s, sparse.SparseArray):
+        raise TypeError(
+            "from_pydata_sparse only accepts objects from the `sparse` library; "
+            "see https://sparse.pydata.org"
+        )
+    if s.ndim > 2:
+        raise _GraphblasException("m.ndim must be <= 2")
```

Comment:
Btw, this isn't covered. Can we add a simple test for this (empty `sparse` object is fine)? See:

https://app.codecov.io/gh/python-graphblas/python-graphblas/pull/347

Also, I'm not sure I like `GraphblasException` here and elsewhere (i.e., `from_numpy` has the same check and exception), but we don't need to change it in this PR.

Reply:
You can also see code coverage here: https://coveralls.io/builds/56178659

The first comment in the PR by coveralls automatically updates to point to the latest coverage for the PR. I find it handy when writing tests.